File: sticky-layoutmanager/src/main/java/androidx/recyclerview/widget/StickyHeaders.java
package androidx.recyclerview.widget;

import androidx.recyclerview.widget.RecyclerView;

/**
 * Created by jay on 2017/12/4 上午10:52
 *
 * Adds sticky headers capabilities to the {@link RecyclerView.Adapter}. Should return {@code true} for all
 * positions that represent sticky headers.
 *
 * @link https://github.com/Doist/RecyclerViewExtensions/blob/master/StickyHeaders
 */
public interface StickyHeaders {

    boolean isStickyHeader(int position);

    interface OnViewAttachListener {
        /**
         * Adjusts any necessary properties of the {@code holder} that is being used as a sticky header.
         *
         * {@link #onStickyHeaderViewDetachedFromWindow(RecyclerView.ViewHolder)} will be called sometime after this method
         * and before any other calls to this method go through.
         * @param holder
         */
        void onStickyHeaderViewAttachedToWindow(RecyclerView.ViewHolder holder);

        /**
         * Reverts any properties changed in {@link #onStickyHeaderViewAttachedToWindow(RecyclerView.ViewHolder)}.
         *
         * Called after {@link #onStickyHeaderViewAttachedToWindow(RecyclerView.ViewHolder)}.
         * @param holder
         */
        void onStickyHeaderViewDetachedFromWindow(RecyclerView.ViewHolder holder);
    }
}

File: sticky-layoutmanager/src/main/java/com/jay/widget/StickyHeaders.kt
package com.jay.widget

import androidx.recyclerview.widget.RecyclerView.ViewHolder

/**
 *
 * Adds sticky headers capabilities to the [RecyclerView.Adapter]. Should return `true` for all
 * positions that represent sticky headers.
 *
 * @link https://github.com/Doist/RecyclerViewExtensions/blob/master/StickyHeaders
 */
interface StickyHeaders {

    fun isStickyHeader(position: Int): Boolean

    interface OnViewAttachListener {
        /**
         * Adjusts any necessary properties of the `holder` that is being used as a sticky header.
         *
         * [.onStickyHeaderViewDetachedFromWindow] will be called sometime after this method
         * and before any other calls to this method go through.
         * @param holder
         */
        fun onStickyHeaderViewAttachedToWindow(holder: ViewHolder)

        /**
         * Reverts any properties changed in [.onStickyHeaderViewAttachedToWindow].
         *
         * Called after [.onStickyHeaderViewAttachedToWindow].
         * @param holder
         */
        fun onStickyHeaderViewDetachedFromWindow(holder: ViewHolder)
    }
}